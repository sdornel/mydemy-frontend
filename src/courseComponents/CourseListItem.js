import React from 'react';
import { connect } from "react-redux";
import { Card, Icon, Image, Grid } from 'semantic-ui-react'
import { Link, Route } from "react-router-dom"

const CourseListItem = (props) => {
    return (
        <div className="course-card-div">
            <Link to={`/course-list/${props.course.id}/`} >
                <Card.Group>
                    <Card className="course-card">
                        <div className="course-card-image-div">
                            <Image classname="course-card-image" src={props.course.picture} />
                        </div>
                        <Card.Content>
                            <h3 className="course-card-text course-card-info">{props.course.name}</h3>
                            <h5 className="course-card-text course-card-info">{props.course.subject}</h5>
                            <h5 className="course-card-text course-card-info-last">{props.course.company.name}</h5>
                        </Card.Content>
                    </Card>
                </Card.Group>
            </Link>
        </div>
    )
}

export default CourseListItem