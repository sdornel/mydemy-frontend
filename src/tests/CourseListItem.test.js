import React from 'react';
import {shallow} from 'enzyme';
import CourseListItem from '../courseComponents/CourseListItem';

const props = {
    course: {
        id: 5000,
        name: "Learn C",
        image: "C Coding Picture",
        subject: "Computer Science",
        company: {
            name: "MIT"
        }
    }
}

describe('CourseListItem', () => {
    it("should receive props", () => {
        const wrapper = shallow(<CourseListItem {...props} />)
        expect(wrapper).toMatchSnapshot()
    })
})