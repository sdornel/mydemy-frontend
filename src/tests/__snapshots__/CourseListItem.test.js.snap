// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseListItem should receive props 1`] = `ShallowWrapper {}`;
